### ISSUE TITLE
Table(react-aria-components): Cannot use Render Props with `Cell`

### ISSUE THREAD

**Issue by jaknas** [adobe/react-spectrum#4408]
# 🐛 Bug Report 

When using render props with Cell, any content returned from render prop function won't be rendered.

## 🤔 Expected Behavior

It should be possible to use render props with `Cell` component and have the returned component be rendered.

## 😯 Current Behavior

as above

## 💻 Code Sample

https://codesandbox.io/s/dazzling-shtern-urznq3?file=/src/App.js

In the codesandbox above, I'd expect `Games` text to be rendered. For comparison, I used render props with column and it works as expected there.

**Comment by LFDanLu** [comment]
I actually think this is in a similar vein to https://github.com/adobe/react-spectrum/issues/4369, https://github.com/adobe/react-spectrum/blob/b92f6029e71aa29a9f7b35e403586856a90004ad/packages/react-aria-components/src/Table.tsx#L845 should have `renderProps.children` instead.
